### PR TITLE
Improved One Way Frequencies dialog, added table save and extra parameters for 'frq' function

### PR DESCRIPTION
**Pull request by Gioche6**
Partly fixes Issue #6670 

@africanmathsinitiative/developers , this is ready for review
@rdstern , @Patowhiz This is ready for review

- I have added the `min.frq` parameter in the `frq` function for the One Way Frequency Dialog. Additionally, I have added the `show.na` parameter in the same function. 

- A function `as.data.frame` has been added converting the result into a data frame

- A save control has been added, to save the resulting data frame for the table and Both options in the dialog.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Gioche6 I liked the way you wrote the PR description, thank you

**Review by lloyddewit — APPROVED** [review]
@Gioche6 This looks good to me

**Comment by lloyddewit** [comment]
@rdstern Please could you test? thanks

**Comment by Patowhiz** [comment]
@lloyddewit @rdstern please note that we found another bug described in issue #6729. Our intention is to solve it in this PR too because we can't finish the items stated above without fixing the bug.

**Review by rdstern — CHANGES_REQUESTED** [review]
This is an excellent start.  There is quite a bit to check here.
a) I get a developer error when I return to the dialogue.  That seems to happen on some dialogues when the settings are not quite right.  
Then I have yet to try saving a table.  That's because it now saves every time, and instead of giving me the results in the output window.  The default is not to save, and I don't think we really need a default last table object saved each time.  So, whn the Save Option isn't checked it should give either html, or simple results in a browser or the output window as before.
Try with the "usual" survey data.
I think the minimum frequencies is working ok.  I'll be able to check more once it displays - as it did before - in the output window.
When it does save, please chack it only saves the results.  There is an extra line on occasions, and whn you set the min.frq then there are 2 extra lines.

**Comment by Gioche6** [comment]
@rdstern , I have made the changes Requested  on the One way frequency Dialog. It is now ready for review.

**Review by rdstern — CHANGES_REQUESTED** [review]
@Gioche6 this is now working well.
I have realised, however, that another option would be useful
Could you add a label for a group box Show Missing Frequencies There are 3 radio buttons, namely True, False and Auto.  The default is True.  If the others are chosen, then it generates the option show.na = FALSE or show.na = "auto".  -

**Comment by Gioche6** [comment]
@rdstern ,  `show.string` is  Logical parameter, if `TRUE`, frequency tables for character vectors will not be printed. This is useful when printing frequency tables of all variables from a data frame, and due to computational reasons character vectors should not be printed

see the example below for a one way frequency summary of a string Variable.

On this first image, ihave set show.string = TRUE,  and the results are as shown.
string1
 
On the second image, I have set show.string =FALSE, for the same. The results are as follows. 

string2

**Comment by rdstern** [comment]
Aha, thank you for the explanation.  That could be reworded more clearly for us as "Omit Character Variables".

It isn't particularly important, for us, because would usually simply not select the character variables if we don't want them.  But it does no harm, so perhaps reword the label and then keep it as it is.

Then I am also happy with the NA stuff within the options, but we need to allow for all 3 options, possibly by the radio buttons.

**Comment by Gioche6** [comment]
@rdstern , I have added the group box nd made necessary change on the design . This is now ready for review and testing.

**Review by rdstern — APPROVED** [review]

**Comment by Gioche6** [comment]
@lloyddewit , could you please review this PR, I think it all better now and can be merged.

**Review by lloyddewit — APPROVED** [review]

File: instat/sdgOneWayFrequencies.vb [hunk 1]
```diff
@@ -36,7 +36,13 @@ Public Class sdgOneWayFrequencies
 
         'Table Only
         ucrChkShowStrings.SetParameter(New RParameter("show.strings", 7), bNewChangeParameterValue:=True, bNewAddRemoveParameter:=True, strNewValueIfChecked:="TRUE", strNewValueIfUnchecked:="FALSE")
-        ucrChkShowStrings.SetText("Show Strings")
+        ucrChkShowStrings.SetText("Omit Character Variabless")
```
Comment (shadrackkibet) [inline instat/sdgOneWayFrequencies.vb:39]: Typo? `Variabless`

Comment (Gioche6) [reply instat/sdgOneWayFrequencies.vb:39]: @shadrackkibet , let me quickly create a new PR , correcting the typo. Thank you for pointing it out

**Pull request merged**